src/kernels/raw/launcher/kernelEnvVarsService.node.ts

```diff
@@ -96,52 +95,68 @@ export class KernelEnvironmentVariablesService {
+            interpreterEnv = interpreterEnv || {};
+
+            if (this.configService.getSettings(resource).useOldKernelResolve) {
+                this.envVarsService.mergeVariables(interpreterEnv, mergedVars); // interpreter vars win over proc.
+                this.envVarsService.mergeVariables(kernelEnv, mergedVars); // kernels vars win over interpreter.
+                this.envVarsService.mergeVariables(customEnvVars, mergedVars); // custom vars win over all.
+                // Reinitialize the PATH variables.
+                // The values in `PATH` found in the interpreter trumps everything else.
+                // If we have more PATHS, they need to be appended to this PATH.
+                // Similarly for `PTYHONPATH`
+                // Additionally the 'PATH' variable may have different case in each, so account for that.
+                let otherEnvPathKey = Object.keys(interpreterEnv).find((k) => k.toLowerCase() == 'path');
+                const processPathKey =
+                    Object.keys(mergedVars).find((k) => k.toLowerCase() == 'path') || otherEnvPathKey;
+                if (otherEnvPathKey && processPathKey) {
+                    mergedVars[processPathKey] = interpreterEnv[otherEnvPathKey];
+                }
+                if (interpreterEnv['PYTHONPATH']) {
+                    mergedVars['PYTHONPATH'] = interpreterEnv['PYTHONPATH'];
+                }
+                otherEnvPathKey = Object.keys(customEnvVars).find((k) => k.toLowerCase() == 'path');
+                if (otherEnvPathKey && customEnvVars[otherEnvPathKey]) {
+                    this.envVarsService.appendPath(mergedVars, customEnvVars[otherEnvPathKey]!);
+                }
+                otherEnvPathKey = Object.keys(kernelEnv).find((k) => k.toLowerCase() == 'path');
+                if (otherEnvPathKey && kernelEnv[otherEnvPathKey]) {
+                    this.envVarsService.appendPath(mergedVars, kernelEnv[otherEnvPathKey]!);
+                }
+                if (customEnvVars.PYTHONPATH) {
+                    this.envVarsService.appendPythonPath(mergedVars, customEnvVars.PYTHONPATH);
+                }
+                if (kernelEnv.PYTHONPATH) {
+                    this.envVarsService.appendPythonPath(mergedVars, kernelEnv.PYTHONPATH);
+                }
+                // Ensure the python env folder is always at the top of the PATH, this way all executables from that env are used.
+                // This way shell commands such as `!pip`, `!python` end up pointing to the right executables.
+                // Also applies to `!java` where java could be an executable in the conda bin directory.
+                if (interpreter) {
+                    this.envVarsService.prependPath(mergedVars, path.dirname(interpreter.uri.fsPath));
+                }
+            } else {
+                Object.assign(mergedVars, interpreterEnv, kernelEnv); // kernels vars win over interpreter.
```

Comment:
With the new Python API we can just use their API as is to get the resolve variables and merge the kernel variables